test/lit/passes/precompute-effects.wast

```diff
@@ -0,0 +1,203 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+
+;; RUN: wasm-opt %s --remove-unused-names --precompute-propagate --fuzz-exec -all -S -o - \
+;; RUN:   | filecheck %s
+
+(module
+ ;; CHECK:      (global $g (mut i32) (i32.const 10))
+ (global $g (mut i32) (i32.const 10))
+
+ ;; CHECK:      (func $loop (type $1)
+ ;; CHECK-NEXT:  (local $temp i32)
+ ;; CHECK-NEXT:  (local.set $temp
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $loop
+  (local $temp i32)
+  ;; We should not try to precompute this loop. If we attempted to replace it
+  ;; with its children, we'd need to handle the effects of chidren properly,
+  ;; which we do not do in this pass.
+  (loop
+   (local.set $temp
+    (i32.const 10)
+   )
+  )
+ )
+
+ ;; CHECK:      (func $local.set (type $1)
+ ;; CHECK-NEXT:  (local $temp i32)
+ ;; CHECK-NEXT:  (local.set $temp
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (local.tee $temp
+ ;; CHECK-NEXT:    (i32.const 20)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $local.set
+  (local $temp i32)
+  ;; We should not try to precompute a set or tee.
+  (local.set $temp
+   (i32.const 10)
+  )
+  (drop
+   (local.tee $temp
+    (i32.const 20)
+   )
+  )
+ )
+
+ ;; CHECK:      (func $global.set (type $1)
+ ;; CHECK-NEXT:  (global.set $g
+ ;; CHECK-NEXT:   (i32.const 20)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $global.set
+  ;; We should not try to precompute a global set.
+  (global.set $g
+   (i32.const 20)
+  )
+ )
+
+ ;; CHECK:      (func $binary-tee (type $0) (result i32)
+ ;; CHECK-NEXT:  (local $temp i32)
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (local.tee $temp
+ ;; CHECK-NEXT:    (i32.const 10)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (i32.const 20)
+ ;; CHECK-NEXT: )
+ (func $binary-tee (result i32)
```

Comment:
Let's also test the case with a single tee in the right hand side.

Reply:
Done.